lib/services/local/trust.go

```diff
@@ -54,26 +60,51 @@ func (s *CA) DeleteAllCertAuthorities(caType types.CertAuthType) error {
 
 // CreateCertAuthority updates or inserts a new certificate authority
 func (s *CA) CreateCertAuthority(ctx context.Context, ca types.CertAuthority) error {
-	if err := services.ValidateCertAuthority(ca); err != nil {
-		return trace.Wrap(err)
-	}
-	value, err := services.MarshalCertAuthority(ca)
-	if err != nil {
-		return trace.Wrap(err)
-	}
-	item := backend.Item{
-		Key:     backend.Key(authoritiesPrefix, string(ca.GetType()), ca.GetName()),
-		Value:   value,
-		Expires: ca.Expiry(),
-	}
+	return s.CreateCertAuthorities(ctx, ca)
+}
 
-	_, err = s.Create(ctx, item)
+// CreateCertAuthorities creates multiple cert authorities atomically.
+func (s *CA) CreateCertAuthorities(ctx context.Context, cas ...types.CertAuthority) error {
+	var condacts []backend.ConditionalAction
+	var clusterNames []string
+	for _, ca := range cas {
+		if !slices.Contains(clusterNames, ca.GetName()) {
+			clusterNames = append(clusterNames, ca.GetName())
+		}
+		if err := services.ValidateCertAuthority(ca); err != nil {
+			return trace.Wrap(err)
+		}
+
+		value, err := services.MarshalCertAuthority(ca)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		condacts = append(condacts, []backend.ConditionalAction{
+			{
+				Key:       activeKey(ca.GetID()),
+				Condition: backend.NotExists(),
+				Action: backend.Put(backend.Item{
+					Value:   value,
+					Expires: ca.Expiry(),
```

Comment:
Shouldn't we pass along ID and revision too, just in case we're on a backend in mirror mode?